contracts/leaser/src/migrate.rs

```diff
@@ -2,101 +2,305 @@ use lease::api::MigrateMsg;
+        match may_leases_nb {
+            Err(err) => Some(Err(err.into())),
+            Ok(leases_nb) => {
+                if leases_nb <= self.leases_left {
+                    self.leases_left -= leases_nb;
+                    customer.leases.find_map(|lease| {
+                        self.msgs
+                            .schedule_migrate_wasm_no_reply(&lease, MigrateMsg {}, self.new_code_id)
+                            .map(|()| None)
+                            .map_err(Into::into)
+                            .transpose()
+                    })
+                } else {
+                    Some(Ok(customer.customer))
+                }
+            }
+        }
     }
 }
 
-impl TryFrom<MigrateBatch> for Batch {
-    type Error = ContractError;
-    fn try_from(this: MigrateBatch) -> Result<Self, Self::Error> {
-        this.may_batch
+impl From<MigrateBatch> for Batch {
+    fn from(this: MigrateBatch) -> Self {
+        this.msgs
     }
 }
 
 #[cfg(test)]
 mod test {
+    use std::vec::IntoIter;
+
     use lease::api::MigrateMsg;
-    use platform::batch::Batch;
     use sdk::cosmwasm_std::Addr;
 
-    use crate::ContractError;
+    use crate::{
+        migrate::{Customer, MigrationResult},
+        result::ContractResult,
+        ContractError,
+    };
+
+    const LEASE1: &str = "lease1";
+    const LEASE21: &str = "lease21";
+    const LEASE22: &str = "lease22";
+    const LEASE3: &str = "lease3";
+    const LEASE41: &str = "lease41";
+    const LEASE42: &str = "lease42";
+    const LEASE43: &str = "lease43";
+    const CUSTOMER_ADDR1: &str = "customer1";
+    const CUSTOMER_ADDR2: &str = "customer2";
+    const CUSTOMER_ADDR3: &str = "customer3";
+    const CUSTOMER_ADDR4: &str = "customer4";
 
     #[test]
     fn no_leases() {
+        use std::array::IntoIter;
         let new_code = 242;
-        let no_leases = vec![];
+        let no_leases: Vec<Customer<IntoIter<Addr, 0>>> = vec![];
         assert_eq!(
-            Ok(Batch::default()),
-            super::migrate_leases(no_leases.into_iter().map(Ok), new_code)
+            Ok(MigrationResult::default()),
+            super::migrate_leases(no_leases.into_iter().map(Ok), 2, new_code)
+        );
+    }
+
+    #[test]
+    fn more_than_max_leases() {
+        let new_code: u64 = 242;
+        let lease1 = Addr::unchecked(LEASE1);
+        let lease2 = Addr::unchecked(LEASE21);
+        let lease3 = Addr::unchecked(LEASE22);
+        let customer_addr1 = Addr::unchecked(CUSTOMER_ADDR1);
+        let cust1 = Customer::from(
+            customer_addr1.clone(),
+            vec![lease1, lease2, lease3].into_iter(),
         );
+
+        let customers = vec![Ok(cust1)];
+        {
+            let exp = MigrationResult {
+                next_customer: Some(customer_addr1),
+                ..Default::default()
+            };
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(customers.into_iter(), new_code, 2)
+            );
```

Comment:
Expected expression is supposed to be the second argument to the macro.
```suggestion
            assert_eq!(
                super::migrate_leases(customers.into_iter(), new_code, 2),
                Ok(exp)
            );
```

Reply:
same